### ISSUE TITLE
Discussion: Should we add a Copy button for code blocks?

### ISSUE THREAD

**Issue by macdonst** [architect/arc.codes#408]
Would folks find it useful if code blocks had a copy button?

For example when you need to type a terminal command like:

```bash
npm i -g @architect/architect aws-sdk
```

**Comment by tbeseda** [comment]
I definitely like this feature.
I have these items bookmarked from a recent implementation
* https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Interact_with_the_clipboard
* https://stackoverflow.com/questions/400212/how-do-i-copy-to-the-clipboard-in-javascript
Especially now that this sort of things doesn't require a Flash fallback 😱

**Comment by macdonst** [comment]
Yeah, it's pretty easy now. You have a hidden textarea that has the same contents as the pre element. Then you execute a bit of JS on the copy button click.

```javascript
document.querySelector("#textarea").select();
document.execCommand("copy");
```
